probatus/feature_elimination/feature_elimination.py

```diff
@@ -1,26 +1,32 @@
-from probatus.utils import assure_pandas_df, shap_calc, assure_list_of_strings, calculate_shap_importance, \
-    NotFittedError
+from probatus.utils import assure_pandas_df, shap_calc, calculate_shap_importance, \
+    NotFittedError, assure_pandas_series
 import numpy as np
 import matplotlib.pyplot as plt
 import pandas as pd
-from sklearn.model_selection import RandomizedSearchCV, GridSearchCV
 import warnings
+from sklearn.model_selection import RandomizedSearchCV, GridSearchCV, check_cv
+from sklearn.base import clone, is_classifier
+from sklearn.metrics import check_scoring
+from joblib import Parallel, delayed
 
 class ShapRFECV:
     """
-    This class performs Backwards Recursive Feature Elimination, using SHAP features importance. At each round, for a given
-     features set, starting from all available features, a model is optimized (e.g. using RandomSearchCV) and trained.
-     At the end of each round, the n lowest SHAP feature importance features are removed and the model results are
-     stored. The user can plot the performance of the model for each round, and select the optimal number of features
-     and the features set.
-
-    The functionality is similar to [sklearn.feature_selection.RFECV](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.RFECV.html),
-     yet, it removes the lowest importance features based on SHAP features importance and optimizes the hyperparameters
-     of the model at each round.
+    This class performs Backwards Recursive Feature Elimination, using SHAP feature importance. At each round, for a
+        given feature set, starting from all available features, Cross-Validation in order to estimate the SHAP feature
+        importance. For each CV iteration, the model is fitted on n-1 folds, and the SHAP values are computed for the
+        last fold. Later, the SHAP values are combined for all test folds, and the `step` lowest SHAP feature importance
+        features are removed. At the end of the process, the user can plot the performance of the model for each
+        iteration, and select the optimal number of features and the features set.
+
+    The functionality is similar to sklearn.feature_selection.RFECV, yet, it removes the lowest importance features
+        based on SHAP features importance. It also supports the use of GridSearchCV and RandomizedSearchCV passed as a
+        clf, thanks to which you can perform hyperparameter optimization at each step of the search. hyperparameters of
```

Comment:
broken sentence. Also, I think the intro "this class is similar to sklearn RFECV" is useful and might be at the top of the docstring.

Reply:
I would leave it below the description on what it does.